### ISSUE TITLE
Pandas deprecation warning

### ISSUE THREAD

**Issue by uzala92** [ranaroussi/yfinance#1837]
### Describe bug

venv/lib/python3.10/site-packages/yfinance/utils.py:775: FutureWarning: The 'unit' keyword in TimedeltaIndex construction is deprecated and will be removed in a future version. Use pd.to_timedelta instead.
  df.index += _pd.TimedeltaIndex(dst_error_hours, 'h')

venv/lib/python3.10/site-packages/yfinance/base.py:291: FutureWarning: 'T' is deprecated and will be removed in a future version, please use 'min' instead.

### Simple code that reproduces your problem

N/A

### Debug log

N/A

### `yfinance` version

0.2.36

### Python version

3.10

### Operating system

Linux

**Comment by cemsbr** [comment]
In my case:
```
lib/python3.12/site-packages/yfinance/utils
.py:775: FutureWarning: The 'unit' keyword in TimedeltaIndex construction is deprecated and will be removed 
in a future version. Use pd.to_timedelta instead.                                                           
  df.index += _pd.TimedeltaIndex(dst_error_hours, 'h')
```

**Comment by louiskenyon98** [comment]
I have this issue and it is preventing me using the df.download() function.

**Comment by tecodrive** [comment]
same problem, after upgrade pandas

```
.../.local/lib/python3.10/site-packages/yfinance/utils.py:775

FutureWarning: The 'unit' keyword in TimedeltaIndex construction is deprecated and will be removed
in a future version. Use pd.to_timedelta instead.

df.index += _pd.TimedeltaIndex(dst_error_hours, 'h')
```

**Comment by cemsbr** [comment]
Workaround: add the following before using yfinance (i.e. in `src/__init__.py`):

```python
import warnings
warnings.simplefilter(action="ignore", category=FutureWarning)
```

**Comment by janfrederik** [comment]
> Workaround: add the following before using yfinance (i.e. in `src/__init__.py`):
> 
> ```python
> import warnings
> warnings.simplefilter(action="ignore", category=FutureWarning)
> ```

Maybe best to be just a little bit more specific, so that warnings for the rest of your code are not hidden:
> ```python
> import warnings
> warnings.filterwarnings("ignore", category=FutureWarning, module="yfinance")
> ```

or ultra specific:
> ```python
> import warnings
> warnings.filterwarnings("ignore", message="The 'unit' keyword in TimedeltaIndex construction is deprecated and will be removed in a future version. Use pd.to_timedelta instead.", category=FutureWarning, module="yfinance.utils")
> ```

**Comment by ValueRaider** [comment]
Fixed in `dev` branch #1844.
